rdf4j/src/main/java/org/apache/commons/rdf/rdf4j/impl/RepositoryDatasetImpl.java

```diff
@@ -0,0 +1,192 @@
+		if (includeInferred) { 
+			// We'll need to count them all
+			return stream().count();
+		} 
+		// else: Ask directly
+		try (RepositoryConnection conn = getRepositoryConnection()) {
+			return conn.size();
+		}
+	}
+
+	
+	@Override
+	public void add(BlankNodeOrIRI graphName, BlankNodeOrIRI subject, IRI predicate, RDFTerm object) {
+		Resource context = (Resource) rdf4jTermFactory.asValue(graphName);
+		Resource subj = (Resource) rdf4jTermFactory.asValue(subject);
+		org.eclipse.rdf4j.model.IRI pred = (org.eclipse.rdf4j.model.IRI) rdf4jTermFactory.asValue(predicate);
+		Value obj = rdf4jTermFactory.asValue(object);
+		try (RepositoryConnection conn = getRepositoryConnection()) {
+			conn.add(subj, pred, obj,  context);
+			conn.commit();
+		}
+	}
+
+	@Override
+	public boolean contains(Optional<BlankNodeOrIRI> graphName, BlankNodeOrIRI subject, IRI predicate, RDFTerm object) {
+		Resource subj = (Resource) rdf4jTermFactory.asValue(subject);
+		org.eclipse.rdf4j.model.IRI pred = (org.eclipse.rdf4j.model.IRI) rdf4jTermFactory.asValue(predicate);
+		Value obj = rdf4jTermFactory.asValue(object);
+		Resource[] contexts = asContexts(graphName);
+		
+		try (RepositoryConnection conn = getRepositoryConnection()) {
+			return conn.hasStatement(subj, pred, obj, includeInferred, contexts);
+		}
+	}
+
+	private Resource[] asContexts(Optional<BlankNodeOrIRI> graphName) {
+		Resource[] contexts;
+		if (graphName == null) {
+			// no contexts == any contexts
+			 contexts = new Resource[0];
+		} else {	
+			BlankNodeOrIRI g = graphName.orElse(null);
+			Resource context = (Resource) rdf4jTermFactory.asValue(g);
+			contexts = new Resource[] { context };
+		}
+		return contexts;
+	}
+
+	@Override
+	public void remove(Optional<BlankNodeOrIRI> graphName, BlankNodeOrIRI subject, IRI predicate, RDFTerm object) {
+		Resource subj = (Resource) rdf4jTermFactory.asValue(subject);
+		org.eclipse.rdf4j.model.IRI pred = (org.eclipse.rdf4j.model.IRI) rdf4jTermFactory.asValue(predicate);
+		Value obj = rdf4jTermFactory.asValue(object);
+		Resource[] contexts = asContexts(graphName);
+
+		try (RepositoryConnection conn = getRepositoryConnection()) {
+			conn.remove(subj, pred, obj, contexts);
+			conn.commit();
+		}
+	}
+
+	@Override
+	public Stream<RDF4JQuad> stream() {
+		return stream(null, null, null, null);
+	}
+	
+	@Override
+	public Stream<RDF4JQuad> stream(Optional<BlankNodeOrIRI> graphName, BlankNodeOrIRI subject, IRI predicate, RDFTerm object) {
+		Resource subj = (Resource) rdf4jTermFactory.asValue(subject);
+		org.eclipse.rdf4j.model.IRI pred = (org.eclipse.rdf4j.model.IRI) rdf4jTermFactory.asValue(predicate);
+		Value obj = rdf4jTermFactory.asValue(object);
+		Resource[] contexts = asContexts(graphName);
+		
+		RepositoryConnection conn = getRepositoryConnection();
+		// NOTE: connection will be closed outside by the Iterations.stream()
+		RepositoryResult<Statement> statements = conn.getStatements(subj, pred, obj, includeInferred, contexts);
```

Comment:
The Stream is consumed outside and is closable.. shall we just add this to the javadoc of `RDF4JGraph.stream()` that use of the stream must be closed (e.g. try-with-resources)?  (Note that this is not really true in `ModelGraph`)